core/src/main/java/org/elasticsearch/rest/action/admin/indices/RestUpgradeAction.java

```
@@ -58,32 +57,32 @@ public RestUpgradeAction(Settings settings, RestController controller) {
     }
 
     @Override
-    public Runnable prepareRequest(RestRequest request, RestChannel channel, NodeClient client) throws Exception {
+    public RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
         if (request.method().equals(RestRequest.Method.GET)) {
-            return handleGet(request, channel, client);
+            return handleGet(request, client);
         } else if (request.method().equals(RestRequest.Method.POST)) {
-            return handlePost(request, channel, client);
+            return handlePost(request, client);
         }
-        return () -> {};
+        return channel -> {};
```

Maybe better to just throw an `UnsupportedOperationException` here.